### PR TITLE
[sumologicexporter] Add filter structure to the Sumo Logic exporter

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

In order to provide Sumo Logic [fields](https://help.sumologic.com/Manage/Fields) we need structure which will hold and process that information

**Link to tracking Issue:**

#1498 

**Testing:**

Unit tests

### CODE REVIEW

File: exporter/sumologicexporter/filter.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		metadataRegexes[i] = regex
+	}
+
+	return filter{
+		regexes: metadataRegexes,
+	}, nil
+}
+
+// convertAttributeToString returns string which represents given pdata.AttributeValue
+func (f *filter) convertAttributeToString(value pdata.AttributeValue) string {
```
Comment (bogdandrutu) [inline exporter/sumologicexporter/filter.go:52]: This function looks similar to https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/trace/protospan_translation.go#L114, you may want to reuse that

Comment (sumo-drosiek) [reply exporter/sumologicexporter/filter.go:52]: Definitely this is what I need. Thank you!

Comment (bogdandrutu) [reply exporter/sumologicexporter/filter.go:52]: I can read your message in two ways. 1. What is available in core is not useful, what is implemented here is what I need. 2. What is in core is what I need, I will use that.

Which of the versions are you referring to?

Comment (sumo-drosiek) [reply exporter/sumologicexporter/filter.go:52]: I'm referring to the second version. Sorry for confusion 😃

Comment (sumo-drosiek) [reply exporter/sumologicexporter/filter.go]: Done, please look again

File: exporter/sumologicexporter/filter.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+				return
+			}
+		}
+		returnValue[k] = tracetranslator.AttributeValueToString(v, false)
+	})
+	return returnValue
+}
+
+// GetMetadata builds string which represents metadata in alphabetical order
+func (f *filter) GetMetadata(attributes pdata.AttributeMap) Fields {
```
Comment (bogdandrutu) [inline exporter/sumologicexporter/filter.go:82]: Why public?

File: exporter/sumologicexporter/filter.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		metadataRegexes[i] = regex
+	}
+
+	return filter{
+		regexes: metadataRegexes,
+	}, nil
+}
+
+// filter returns map of strings which matches at least one of the filter regexes
+func (f *filter) filter(attributes pdata.AttributeMap) map[string]string {
```
Comment (bogdandrutu) [inline exporter/sumologicexporter/filter.go:52]: Consider `filterIn` to be symmetric with `filterOut`?

File: exporter/sumologicexporter/filter.go (hunk 2 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/filter.go]: No reason, fixed

File: exporter/sumologicexporter/filter.go (hunk 3 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/filter.go]: That makes sense. Updated

**Review by bogdandrutu — DISMISSED** [review]

File: exporter/sumologicexporter/filter.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	"go.opentelemetry.io/collector/consumer/pdata"
+	tracetranslator "go.opentelemetry.io/collector/translator/trace"
+)
+
+type filter struct {
+	regexes []*regexp.Regexp
+}
+
+// Fields represents concatenated metadata
+type Fields string
```
Comment (bogdandrutu) [inline exporter/sumologicexporter/filter.go:32]: Should this be an exported type?

Comment (sumo-drosiek) [reply exporter/sumologicexporter/filter.go]: I don't plan to use it outside of the exporter, so it shouldn't. Fixed

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**